filip/clients/ngsi_v2/cb.py

```diff
@@ -1386,6 +1390,137 @@ def post_command(self,
             self.log_error(err=err, msg=msg)
             raise
 
+    def does_entity_exists(self,
+                           entity_id: str,
+                           entity_type: str) -> bool:
+        """
+        Test if an entity with given id and type is present in the CB
+        Args:
+            entity_id: Entity id
+            entity_type: Entity type
+        Returns:
+            bool; True if entity exists
+
+        Raises:
+            RequestException, if any error occurres (e.g: No Connection),
+            except that the entity is not found
+        """
+        try:
+            self.get_entity(entity_id=entity_id, entity_type=entity_type)
+        except requests.RequestException as ex:
+            # couldn't find a better way to extract error code
+            if not str(ex)[0:36] == "404 Client Error: Not Found for url:":
+                raise
+            return False
+        
+        return True
+
+    def patch_entity(self,
+                     entity: ContextEntity,
+                     old_entity: Optional[ContextEntity] = None) -> None:
+        """
+           Takes a given entity and updates the state in the CB to match it.
+           Args:
```

Comment:
Empty line is missing

Reply:
In which line the empty line should be inserted?, I do not find a violation of the style guide  (Except the now fixed over-indent)